File: 0682-baseball-game/0682-baseball-game.cpp
class Solution {
public:
    int calPoints(vector<string>& operations) {
        stack<int>st;

        for(string op : operations){
            if(op == "D"){
                st.push(st.top()*2);
            }else if(op == "C"){
                st.pop();
            }else if(op == "+"){
                int a = st.top();
                st.pop();
                int b = st.top();

                st.push(a);
                st.push(a + b);
            }else{
                st.push(stoi(op));
            }
        }

        int score = 0;
        while(!st.empty()){
            score += st.top();
            st.pop();
        }

        return score;
    }
};

File: 0682-baseball-game/0682-baseball-game.java
class Solution {
    public int calPoints(String[] operations) {
        Stack<Integer>st = new Stack<>();

        for(String op : operations){
            // "POP" top element and double it, and push it into stack..
            if(op.equals("D")){
                st.push(st.peek()*2);
            }else if(op.equals("C")){ // Simpily "POP" it from the stack
                st.pop();
            }else if(op.equals("+")){ // "POP" top 2 elements & push back 2 elements & result 
                int first = st.pop();
                int second = st.peek();
                st.push(first);
                st.push(first + second);
            }else{                  // It means pointing to number push it
                st.push(Integer.parseInt(op)); // This is for converting "string into integer"
            }
        }
        int score = 0;
        
        while(!st.isEmpty()){       // Always keep storing the top element and pop it as well
            score += st.pop();
        }

        return score;
    }
}